### PR TITLE
github support

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
- Rework github implementation.
- Add tests for github.
- Add API for file content.
- Correct Pagure tests.

### CODE REVIEW

File: ogr/abstract.py [hunk 1]
```diff
@@ -267,6 +267,16 @@ def change_token(self, new_token: str):
         """
         raise NotImplementedError
 
+    def get_file_content(self, path: str, revision="master") -> Optional[bytes]:
+        """
+        Get a content of the file in the repo.
+
+        :param revision: branch or commit (defaults to master)
+        :param path: str
+        :return: bytes or None if there is no such file
```
Comment (TomasTomecek) [inline ogr/abstract.py]: please raise an Exception if the file does not exist

Comment (TomasTomecek) [reply ogr/abstract.py]: and returning str will be easier for everyone

File: ogr/services/our_pagure.py [hunk 2]
```diff
@@ -20,15 +20,17 @@ def api_url(self):
     def repo_name(self):
         return self.repo.split("/")[1]
 
-    def get_api_url(self, *args, add_fork=True):
+    def get_api_url(self, *args, add_fork=True, api_url=True):
```
Comment (TomasTomecek) [inline ogr/services/our_pagure.py]: the api_url argument is extremely confusing

**Comment by lachmanfrantisek** [comment]
@TomasTomecek fixed

**Comment by TomasTomecek** [comment]
LGTM, nicely done!

**Pull request merged**